ambari-server/src/main/java/org/apache/ambari/server/security/ldap/AmbariLdapDataPopulator.java

```diff
@@ -447,10 +447,17 @@ protected void refreshGroupMembers(LdapBatchDto batchInfo, LdapGroupDto group, M
    * @return the set of LDAP groups for the given name
    */
   protected Set<LdapGroupDto> getLdapGroups(String groupName) {
+    String baseDn = ldapServerProperties.getBaseDN().toLowerCase();
+    String groupBase = ldapServerProperties.getGroupBase().toLowerCase();
+
+    //If groupBase is set incorrectly or isn't set - search in BaseDn
+    int indexOfBaseDn = groupBase.indexOf(baseDn);
+    groupBase = indexOfBaseDn <= 0 ? "" : groupBase.substring(0, indexOfBaseDn - 1);
```

Comment:
It is not clear what you are doing here.  If the groupBase contains the baseDn, then strip off the baseDN, else groupBase is an empty string?

Also, what happens if groupBase is not set... you will get a NPE.

Reply:
Actually, I copied/pasted code from Ambari itself ^^ from https://github.com/apache/ambari/blob/trunk/ambari-server/src/main/java/org/apache/ambari/server/security/authorization/AmbariLdapBindAuthenticator.java#L276